maggy/src/services/executor.py

```
@@ -0,0 +1,253 @@
+                *cmd,
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.STDOUT,
+                cwd=working_dir,
+            )
+            stdout, _ = await asyncio.wait_for(proc.communicate(), timeout=600)
+            return (stdout or b"").decode("utf-8", errors="replace")
+        except asyncio.TimeoutError:
+            return "ERROR: claude timed out after 10 minutes"
+        except FileNotFoundError:
+            return "ERROR: `claude` CLI not found on PATH. Install Claude Code first."
+        except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)